### PR TITLE
Patch bug in call signature, add pylint

### PR DESCRIPTION
**Pull request by hancush**
## Description

Hi, @jamesturk – Thanks for this most excellent library! We use it all the time at @datamade. 

This PR patches a bug caused by calling `key_for_request` with the wrong signature. It also adds pylint, which will catch this type of error.

Example stack trace prior to this patch from our trusty scraper:

```
Traceback (most recent call last):
  File "/usr/local/bin/pupa", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/pupa/cli/__main__.py", line 75, in main
    subcommands[args.subcommand].handle(args, other)
  File "/usr/local/lib/python3.10/site-packages/pupa/cli/commands/update.py", line 334, in handle
    return self.do_handle(args, other, juris)
  File "/usr/local/lib/python3.10/site-packages/pupa/cli/commands/update.py", line 391, in do_handle
    report["scrape"] = self.do_scrape(juris, args, scrapers)
  File "/usr/local/lib/python3.10/site-packages/pupa/cli/commands/update.py", line 215, in do_scrape
    report[scraper_name] = scraper.do_scrape(**scrape_args)
  File "/usr/local/lib/python3.10/site-packages/pupa/scrape/base.py", line 120, in do_scrape
    for obj in self.scrape(**kwargs) or []:
  File "/app/lametro/bills.py", line 204, in scrape
    for matter in matters:
  File "/app/lametro/bills.py", line 166, in matters
    for api_event, _ in event_scraper.events(since_datetime=since_datetime):
  File "/usr/local/lib/python3.10/site-packages/legistar/events.py", line 278, in events
    web_event = self._get_web_event(api_event)
  File "/usr/local/lib/python3.10/site-packages/legistar/events.py", line 381, in _get_web_event
    return self.web_detail(api_event)
  File "/usr/local/lib/python3.10/site-packages/legistar/events.py", line 391, in web_detail
event_page = self._webscraper.lxmlize(insite_url)
  File "/usr/local/lib/python3.10/site-packages/legistar/base.py", line 89, in lxmlize
    response = self.get(url, verify=False)
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/scrapelib/__init__.py", line 574, in request
    resp = super().request(
  File "/usr/local/lib/python3.10/site-packages/scrapelib/__init__.py", line 389, in request
    request_key = self.key_for_request(method, url, params)
TypeError: LegistarEventsScraper.key_for_request() takes 3 positional arguments but 4 were given; 27459)
```

Separately, pylint caught an error caused by the release of urllib3 2.0 (see: https://github.com/psf/requests/issues/6443), so I've also capped that dependency.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,12 +23,14 @@ classifiers=[
 scrapeshell = "scrapelib.__main__:scrapeshell"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = ">=3.7.2,<4.0"
```
Comment (hancush) [inline pyproject.toml:26]: This range was recommended by poetry when I added pylint. Ok by you?

File: pyproject.toml [hunk 2]
```diff
@@ -23,12 +23,14 @@ classifiers=[
...
 requests = {extras = ["security"], version = "^2.28.1"}
+urllib3 = "<2"
```
Comment (hancush) [inline pyproject.toml:28]: Reference: https://github.com/psf/requests/issues/6443

File: scrapelib/__init__.py [hunk 3]
```diff
@@ -390,7 +391,7 @@ def request(  # type: ignore
 
         method = method.lower()
 
-        request_key = self.key_for_request(method, url, params)
+        request_key = self.key_for_request(method, url, params=params)
```
Comment (hancush) [inline scrapelib/__init__.py:394]: Bugfix.

File: scrapelib/__init__.py [hunk 4]
```diff
@@ -543,9 +544,9 @@ def request(  # type: ignore
         # for example 'HIGH:!DH:!aNULL' to bypass "dh key too small" error
         # https://stackoverflow.com/questions/38015537/python-requests-exceptions-sslerror-dh-key-too-small
         if ciphers_list_addition:
-            requests.packages.urllib3.util.ssl_.DEFAULT_CIPHERS += ciphers_list_addition  # type: ignore
+            urllib3.util.ssl_.DEFAULT_CIPHERS += ciphers_list_addition  # type: ignore
```
Comment (hancush) [inline scrapelib/__init__.py:547]: `requests.packages.urllib3` is an alias for `urllib3`, so I use the lib directly to shut pylint up. Reference: https://github.com/pylint-dev/pylint/issues/4584

**Comment by jamesturk** [comment]
Hi Hannah, 

Thanks for this!  I'm a bit confused about the `pylint` piece though, it doesn't seem to catch the error:

(on a checkout of main, with `pylint` installed)

- `poetry run pylint -E scrapelib` returns no errors
- `poetry run pytlint scrapelib` returns a bunch of pylint noise, but no mention of line 393's error

I'd rather not add `pylint` if it isn't catching this error, `ruff` doesn't seem to either.  It is a tough error for a linter to catch, the only one I'd expect to catch it would be `mypy`.

**Comment by hancush** [comment]
Ahhhhh, complete brain fart. The call signature was our problem, and I got confused with pylint. FWIW, mypy will catch too many positional args if you specify kwargs as keyword only, but that's not the problem here. Opened #222 to pin urllib3, but the rest of this is unnecessary. My bad!

**Pull request closed without merging**